### PR TITLE
562 story create graduation application form component

### LINKED ISSUE #562: Story: Create Graduation Application Form Component
# Create Graduation Application Form Component

As a **developer**, I want **a form component**, so that **when interacting with the lab I can store user information for the graduation information**.

Associated Epic: #490 

## Definition of Done

- [ ] Acceptance criteria defined
- [ ] Solution tasks specified
- [ ] Feature branch created
- [ ] ~~Unit tests written (on API code only for now)~~
- [ ] ~~Solution passes full suite of unit tests~~
- [ ] Design documents updated if needed
- [ ] Pull request created
- [ ] Code is reviewed
- [ ] Code is merged to `develop` branch
- [ ] ~~Tests pass on CI~~
- [ ] ~~Deployed to staging environment~~
- [ ] Acceptance criteria is met

## Acceptance Criteria

- [ ] Given **user**, when **they enter and submit their info to the form**, then **then the user state is persisted in a lab level context**
- [ ] Given **user input**, when **a user submits their responses**, then **there should be string sanitation and validation to make sure that we are protecting against any injection or cross site scripting issues**

## Solution Tasks

- [ ] Create form and style it using tailwind
- [ ] Add and handle the react context and reducers,

### PR DESCRIPTION
**Pull request by apr8760**
Jest testing to come.
To test, add  component to App.js --> "GradApplication path="/Lab12/Exercise/*" user={state.main.user} " and run.

- [ ] No discrepancies across browsers. Eg. chrome vs safari
- [ ] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [ ] CSS (https://jigsaw.w3.org/css-validator/)
- [ ] Google sign-in is operational
- [ ] Accessibility functions
- [ ] Pages can scale without distorting page
- [ ] No dead links
- [ ] Navbar is consistent across the site
- [ ] Pages are screen-reader accessible
- [ ] Contrast meets standards for accessibility
- [ ] Pages are keyboard accessible
- [ ] Passes WAVE Evaluation Tool chrome extension

### CODE REVIEW

**Review by dm9965 — APPROVED** [review]
Would like to see the red error handling message aligned with the input itself, however the functionality is correct. Approved.

Screenshot 2024-04-11 at 3 44 49 PM

**Review by dm9965 — CHANGES_REQUESTED** [review]
Set the red error handling to be aligned with the input box

Screenshot 2024-04-11 at 3 44 49 PM

File: client/src/components/exercise/lab12/GradApplication.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
+/* eslint-disable no-unused-vars */
```
Comment (heathermoses) [inline client/src/components/exercise/lab12/GradApplication.js]: nit: remove this eslint line

**Review by heathermoses — APPROVED** [review]
This looks great! Great job on your first PR! :)

**Review by heathermoses — APPROVED** [review]
I tested this by ensuring the error handling works and making sure tests pass. Looks good

**Pull request merged**